narwhal/worker/src/quorum_waiter.rs

```diff
@@ -106,36 +117,53 @@ impl QuorumWaiter {
                     // the dag). This should reduce the amount of synching.
                     let threshold = self.committee.quorum_threshold();
                     let mut total_stake = self.committee.stake(&self.name);
-                    loop {
-                        tokio::select! {
-                            Some(stake) = wait_for_quorum.next() => {
+
+                    pipeline.push_back(async move {
+                        // A future that sends to 2/3 stake then returns. Also prints an error
+                        // if we terminate before we have managed to get to the full 2/3 stake.
+                        let _ = &wait_for_quorum;
+                        loop{
+                            if let Some(stake) = wait_for_quorum.next().await {
                                 total_stake += stake;
                                 if total_stake >= threshold {
-                                    if self.tx_batch.send(batch).await.is_err() {
-                                        tracing::debug!("{}", DagError::ShuttingDown);
-                                    }
-                                    break;
-                                }
-                            }
-
-                            result = self.rx_reconfigure.changed() => {
-                                result.expect("Committee channel dropped");
-                                let message = self.rx_reconfigure.borrow().clone();
-                                match message {
-                                    ReconfigureNotification::NewEpoch(new_committee)
-                                        | ReconfigureNotification::UpdateCommittee(new_committee) => {
-                                            self.network.cleanup(self.committee.network_diff(&new_committee));
-                                            self.committee = new_committee;
-                                            tracing::debug!("Dropping batch: committee updated to {}", self.committee);
-                                            break; // Don't wait for acknowledgements.
-                                    },
-                                    ReconfigureNotification::Shutdown => return
+                                    break
                                 }
+                            } else {
+                                tracing::error!("Batch dissemination ended without a quorum.");
+                                break;
```

Comment:
I believe leaving this here will eventually send the batch for further processing to downstream components and to primary for proposal inclusion. Actually if that happens - all replied but not enough stake gathered - we are looking to a pathogen of the system (bug). In this case I believe we should either abandon the batch or crash (if that's really severe for us).

Reply:
From this particular location its hard to see what more to do. Adding a TODO about it.